pages/seed/getting-started/data-connection.mdx

```diff
@@ -0,0 +1,192 @@
+        { users_user_message_recipient_idTousers: {} },
+        { users_user_message_recipient_idTousers: {} }
+      ],
+      // Ensure each author creates a new user
+      user_message_user_message_recipient_idTousers: [
+        { users_user_message_author_idTousers: {} },
+        { users_user_message_author_idTousers: {} }
+      ],
+      channel: [{
+        channel_thread: [{
+          channel_thread_message: [{}, {}, {}]
+        }]
+      }]
+    })
+  );
+```
+
+### Plan connect
+Sometimes, you need more control over generated data without defining each element. Returning to our example, we have multiple "pools of users" in our app via members.
+
+1. A workspace connects to a pool of users via `workspace_members`
+2. A channel connects to a pool of users via `channel_members`
+
+We want to ensure:
+1. All authors and channel members in a workspace are a subset of the workspace members.
+2. All thread messages in a channel are authored by channel members.
+3. All direct messages between users are between users sharing at least one workspace.
+
+We'll create pools of users and use plan connections to generate data with the correct relationships:
+
+```ts
+  // Create a pool of between 2 and 10 users for our workspace
+  const { users } = await seed.users((x) => x({min: 2, max: 10}));
+  // Create our workspace and connect all our users to it
+  const { workspace, workspace_member } = await seed.workspace((x) => x(1, () => ({
+    // Each workspace will have between 1 and 10 members
+    workspace_member: (x) => x(users.length)
+  }), 
+  // Connect the workspace members to the pool of users we created
+  { connect: { users }}));
+
+  // Use the workspace and workspace_member objects to create more complex relationships
+  for (const w of workspace) {
+    // Retrieve all members for this specific workspace
+    const workspace_members = workspace_member.filter((m) => m.workspace_id === w.id);
+    // Pick between 1 and the number of workspace members to be channel members
+    const channel_users = copycat.someOf(w.id, [1, workspace_members.length], workspace_members).map((m) => ({ id: m.user_id }));
+    // Create between 1 and 10 channels for this workspace
+    await seed.channel((x) => x({ min: 1, max: 10 }, () => ({
+      // Add all channel users as members of the current channel
+      channel_member: (x) => x(channel_users.length),
+      // Create between 1 and 10 threads for each channel
+      channel_thread: (x) => x({ min: 1, max: 10 }, ({
+        // Create between 0 and 10 messages for each thread
+        channel_thread_message: (x) => x({ min: 0, max: 10 })
+      }))
+    }), {
+      connect: { 
+        // Connect all channels to the current workspace
+        workspace: [w],
+        // Connect all channel users to the pool of users for this channel
+        users: channel_users
+      }
+    }));
+  }
+```
+
+Now, we can generate more workspaces that match our constraints by adjusting one variable.
```

Comment:
I didn't follow this part here. Is it referring to the `seed.workspace((x) => x(1, ...`, which can be adjusted to something else by changing `1` to something else?

Reply:
Yes, maybe I'll extract this into a named variable like `workspacesNumber = 1` so it's easier to connect the dots 🤔